meta-security-smack/recipes-core/systemd/systemd_%.bbappend

```diff
@@ -57,6 +64,62 @@ EXTRA_OECONF_append_smack = " --with-smack-run-label=System"
+    install_file ${D}${systemd_unitdir}/system/systemd-tmpfiles-setup.service.d/smack.conf <<EOF
+[Service]
+ExecStartPost=/bin/sh -c '([ ! -d /var/tmp ] || chsmack -L -a \"*\" /var/tmp) && ([ ! -d /var/log ] || chsmack -L -a System::Log /var/log && chsmack -L -t /var/log)'
+EOF
+
+    # Mount /tmp publicly accessable. Based on patch by Michael Demeter <michael.demeter@intel.com>.
+    # Upstream systemd temporarily had SmackFileSystemRoot for this (https://github.com/systemd/systemd/pull/1664),
+    # but it was removed again (https://github.com/systemd/systemd/issues/1696) because
+    # util-linux mount will ignore smackfsroot when Smack is not active. However,
+    # busybox is not that intelligent.
+    #
+    # When using busybox mount, adding smackfsroot=* and booting without
+    # Smack (i.e. security=none), tmp.mount will fail with an error about
+    # "Bad mount option smackfsroot".
+    install_file ${D}${systemd_unitdir}/system/tmp.mount.d/smack.conf <<EOF
+[Mount]
+Options=smackfsroot=*
+EOF
+
+    # Run systemd-journald with the hat ("^") Smack label.
+    #
+    # The journal daemon needs global read access to gather information
+    # about the services spawned by systemd. The hat label is intended
+    # for this purpose. The journal daemon is the only part of the
+    # System domain that needs read access to the User domain. Giving
+    # the journal daemon the hat label means that we can remove the
+    # System domain's read access to the User domain.
+    #
+    # Original author: Casey Schaufler <casey@schaufler-ca.com>
+    #
+    # This is considered a configuration change and thus distro specific.
+    install_file ${D}${systemd_unitdir}/system/systemd-journald.service.d/smack.conf <<EOF
+[Service]
+SmackProcessLabel=^
+EOF
+}
+
+# Will get installed in ${sysconfdir}/udev/rules.d/ by base systemd recipe.
+SRC_URI += "file://udev-smack-default.rules"
+
+# Maintaining trivial, non-upstreamable configuration changes as patches
+# is tedious. But in same cases (like early mounting of special directories)
+# the configuration has to be in code. We make these changes here directly.
+do_patch[prefuncs] += "patch_systemd"
+do_patch[vardeps] += "patch_systemd"
+patch_systemd() {
+    # Handling of /run and /sys/fs/cgroup. Make /run a transmuting directory to
+    # enable systemd communications with services in the User domain.
+    # Original patch by Michael Demeter <michael.demeter@intel.com>.
+    #
+    # We simplify the patching by touching only lines which check the result of
+    # mac_smack_use(). Those are the ones which are used when Smack is active.
+    #
+    # smackfsroot=* on /sys/fs/cgroup may be upstreamable, but smackfstransmute=System::Run
+    # is too distro specific (depends on Smack rules) and thus has to remain here.
```

Comment:
On Wed, 2016-01-13 at 04:44 -0800, Ismo Puustinen wrote:
> In meta-security-smack/recipes-core/systemd/systemd_%.bbappend:
> 
> > +
> > +# Maintaining trivial, non-upstreamable configuration changes as patches
> > +# is tedious. But in same cases (like early mounting of special directories)
> > +# the configuration has to be in code. We make these changes here directly.
> > +do_patch[prefuncs] += "patch_systemd"
> > +do_patch[vardeps] += "patch_systemd"
> > +patch_systemd() {
> > +    # Handling of /run and /sys/fs/cgroup. Make /run a transmuting directory to
> > +    # enable systemd communications with services in the User domain.
> > +    # Original patch by Michael Demeter <michael.demeter@intel.com>.
> > +    #
> > +    # We simplify the patching by touching only lines which check the result of
> > +    # mac_smack_use(). Those are the ones which are used when Smack is active.
> > +    #
> > +    # smackfsroot=* on /sys/fs/cgroup may be upstreamable, but smackfstransmute=System::Run
> > +    # is too distro specific (depends on Smack rules) and thus has to remain here.
> 
> I would say that maintaining this sed command might get tedious too.
> It's difficult to see at a glance what it does.

True, but I don't see a good alternative other than returning to
patches, which is inevitably going to break again.

Would it help to show the lines that it is expected to modify?